osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -89,112 +132,162 @@ public override double Calculate(Dictionary<string, double> categoryRatings = nu
 
         private double computeAimValue()
         {
-            double rawAim = Attributes.AimStrain;
+            if (Beatmap.HitObjects.Count <= 1)
+                return 0;
 
-            if (mods.Any(m => m is OsuModTouchDevice))
-                rawAim = Math.Pow(rawAim, 0.8);
+            // Get player's throughput according to combo
```

Comment:
"throughput" also requires explanation as I don't understand what it's supposed to mean in this context.

Reply:
"throughput" is a term from fitt's law, basically it's a measure of skill